modules/rostests/apitests/advapi32/RegCreateKeyEx.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for RegCreateKeyExW.
+ * COPYRIGHT:   Copyright 2023 Doug Lyons <douglyons@douglyons.com>
+ */
+
+/*
+ * Idea based loosely on code from the following:
+ * https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--
+ */
+
+#include <apitest.h>
+#include <stdio.h>
+
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+#include <windef.h>
+#include <aclapi.h>
+
+
+START_TEST(RegCreateKeyEx)
+{
+    HKEY hkey_main;
+    DWORD dwRes, dwDisposition;
+    PACL pACL = NULL;
+    PSECURITY_DESCRIPTOR pSD = NULL;
+    PSID pEveryoneSID = NULL, pAdminSID = NULL;
+    SID_IDENTIFIER_AUTHORITY SIDAuthWorld = {SECURITY_WORLD_SID_AUTHORITY};
+    SID_IDENTIFIER_AUTHORITY SIDAuthNT = {SECURITY_NT_AUTHORITY};
+    EXPLICIT_ACCESSW ea[2];
+    SECURITY_ATTRIBUTES sa = { 0 };
+    LONG lRes;
+    BOOL bRes;
+    LONG ErrorCode = 0;
+    HKEY hkSub = NULL;
+
+    // If any of the test keys already exist, delete them to ensure proper testing
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        ErrorCode = RegDeleteKeyW(HKEY_CURRENT_USER, L"mykey");
+        ok_dec(ErrorCode, ERROR_SUCCESS);
+        if (ErrorCode != ERROR_SUCCESS)
+        {
+            skip("'HKCU\\mykey' cannot be deleted. Terminating test\n");
+            goto Cleanup;
+        }
+    }
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey1", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        ErrorCode = RegDeleteKeyW(HKEY_CURRENT_USER, L"mykey1");
+        ok_dec(ErrorCode, ERROR_SUCCESS);
+        if (ErrorCode != ERROR_SUCCESS)
+        {
+            skip("'HKCU\\mykey1' cannot be deleted. Terminating test\n");
+            goto Cleanup;
+        }
+    }
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey2", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        ErrorCode = RegDeleteKeyW(HKEY_CURRENT_USER, L"mykey2");
+        ok_dec(ErrorCode, ERROR_SUCCESS);
+        if (ErrorCode != ERROR_SUCCESS)
+        {
+            skip("'HKCU\\mykey2' cannot be deleted. Terminating test\n");
+            goto Cleanup;
+        }
+    }
+
+    // Setup GetLastError to known value for tests
+    SetLastError(0xdeadbeef);
+
+    // Create a well-known SID for the Everyone group.
+    bRes = AllocateAndInitializeSid(&SIDAuthWorld, 1,
+           SECURITY_WORLD_RID,
+           0, 0, 0, 0, 0, 0, 0,
+           &pEveryoneSID);
+    ok(bRes, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (!bRes)
+        {
+            skip("EveryoneSID not initialized. Terminating test\n");
+            goto Cleanup;
+        }
```

Comment:
indentation is wrong here for this newly added block. Curly braces should be under the i of the if.

Reply:
Fixed. Thanks.